Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.h

```diff
@@ -50,6 +50,8 @@ class WEBCORE_EXPORT ImageBufferIOSurfaceBackend : public ImageBufferCGBackend {
     
     static constexpr RenderingMode renderingMode = RenderingMode::Accelerated;
 
+    bool isImageBufferIOSurfaceBackend() const override;
```

Comment:
We have four backends which are backed by an IOSurface 

1. ImageBufferIOSurfaceBackend
2. ImageBufferShareableMappedIOSurfaceBackend
3. ImageBufferRemoteIOSurfaceBackend
4. ImageBufferShareableMappedIOSurfaceBitmapBackend

Only the first two will return true for this function. Is this what we want? I am just confused by ImageBufferShareableMappedIOSurfaceBackend and I do not whether it should return true or not.

Reply:
I think it would be definitely wrong to say yes from _this_ (the is-castable function).

I also think it's not necessary to make surface() work for the later two, simply because the old mechanism wouldn't have covered them either?